API/RecommendationService/scenario_service.py

```diff
@@ -0,0 +1,128 @@
+                'type': RecommendType.Scenario
+            }
+            if 'description' in item:
+                scenario['reason'] = item['description']
+            result.append(scenario)
+
+    return result[0: top_num]
+
+
+def get_search_results(trigger_commands: List[str], top: int = 5):
+    """Search related sceanrios using cognitive search
+
+    Args:
+        trigger_commands (List[str]): list of commands used to search
+        top (int, optional): top num of returned results. Defaults to 5.
+
+    Returns:
+        list[dict]: searched scenarios
+    """
+    if len(trigger_commands) == 0:
+        return []
+    service_endpoint = os.environ["SCENARIO_SEARCH_SERVICE_ENDPOINT"]
+    search_client = SearchClient(endpoint=service_endpoint,
+                                 index_name=os.environ["SCENARIO_SEARCH_INDEX"],
+                                 credential=AzureKeyCredential(os.environ["SCENARIO_SEARCH_SERVICE_SEARCH_KEY"]))
+
+    search_statement = "(" + " OR ".join([f'"{cmd}"' for cmd in trigger_commands][:-1]) + ") AND " + f'"{trigger_commands[-1]}"'
+    search_statement = f'"{trigger_commands[-1]}" OR ({search_statement})'
+    results = search_client.search(
+        search_text=search_statement,
+        include_total_count=True,
+        search_fields=["commandSet/command"],
+        highlight_fields="commandSet/command",
+        top=top,
+        query_type='full')
+    results = list(results)
+    return results
+
+
+def get_scenario_recommendation_from_search(command_list, top_num=5):
+    """Recommend Scenarios that current context could be in
+
+    Args:
+        command_list (list[str]): commands used to trigger
+        top_num (int, optional): top num of recommended results. Defaults to 5.
+
+    Returns:
+        list[dict]: searched scenarios
+    """
+    if len(command_list) == 0:
+        return []
+    trigger_len = int(os.environ.get("ScenarioRecommendationTriggerLength", "3"))
+    trigger_commands = get_latest_cmd(command_list, trigger_len)
+    trigger_commands = [cmd[3:] if cmd.startswith("az ") else cmd for cmd in trigger_commands]
+    searched = get_search_results(trigger_commands, top_num)
+
+    results = []
+    for item in searched:
+        # get all commands in searched scenario with `az ` stripped
+        cmds = [cmd['command'][3:] for cmd in item['commandSet'] if len(cmd['command']) > 3]
+        # get indices of commands that the user has not executed yet
+        exec_idx = [idx for idx, cmd in enumerate(cmds) if cmd not in trigger_commands]
+        # avoid recommending scenarios where the user has executed their last command
+        if len(cmds) - 1 not in exec_idx:
+            continue
+        scenario = {
+            'scenario': item['name'],
+            'nextCommandSet': strip_az_in_command_set(item['commandSet']),
+            'source': RecommendationSource.Search,
+            'type': RecommendType.Scenario,
+            'execIdx': exec_idx,
```

Comment:
Use full names instead of abbreviations to make fields more readable
```suggestion
            'executeIndex': exec_idx,
```